### DISCUSSION TITLE
Confused by how's C_0, C_1 being calculated in paper

### DISCUSSION THREAD

**Discussion by LittleLittleCloud** [fairlearn/fairlearn#1090] [Q&A] [answered]
Hi fairlearn team, I am reading this [paper](https://arxiv.org/pdf/1803.02453.pdf) and am confused on how C_0, C_1 is calculated for DP in `signed_weight` function

The generic fomular for C_0 and C_1 is

![image](https://user-images.githubusercontent.com/16876986/168402946-bac5a26a-d1c2-44b2-9590-e5495e4f67e1.png)

but for DP, according to paper, it's
![image](https://user-images.githubusercontent.com/16876986/168403038-1b4ebbd4-390a-47f3-8a8b-2afca04046b0.png)

Just wondering how's the second term for C_0 disappear and how's C_1's second term being transform to the given format. 

Regards

**Comment by romanlutz** [comment, marked as answer]
Adding @MiroDudik who is one of the authors of the paper.

If I remember this correctly, the function `g` just evaluates to `h(X)`, so in the `C_0` case that `0`, and in the `C_1` case that's `1`. Hence, the entire term on the right is `0`.

My personal notes from quite some time ago have this little note (which is not from the paper, but rather my derivation of this result, so take it with a grain of salt):
![image](https://user-images.githubusercontent.com/10245648/168703390-aa168d48-8f00-4426-8a0a-c619177d48eb.png)
